python/cudf_polars/cudf_polars/dsl/ir.py

```diff
@@ -1381,7 +1383,14 @@ def do_evaluate(
             names.append(name)
         group_keys, raw_tables = grouper.aggregate(requests)
         results = [
-            Column(column, name=name, dtype=request.value.dtype)
+            Column(
+                plc.unary.cast(column, plc.DataType(plc.TypeId.UINT32))
+                if isinstance(request.value, expr.Agg)
+                and request.value.name == "n_unique"
+                else column,
+                name=name,
+                dtype=schema[name],
+            )
```

Comment:
Yeah, if we insert the cast in the IR during translation, we don't need special-casing here.

Reply:
Thanks done